tachyon/math/base/groups.h

```diff
@@ -76,25 +87,28 @@ class MultiplicativeGroup : public MultiplicativeSemigroup<G> {
     }
 
 #if defined(TACHYON_HAS_OPENMP)
-    using G2 = decltype(std::declval<G>().Inverse());
     size_t thread_nums = static_cast<size_t>(omp_get_max_threads());
     if (size >=
         size_t{1} << (thread_nums / kParallelBatchInverseDivisorThreshold)) {
       size_t chunk_size = base::GetNumElementsPerThread(groups);
       size_t num_chunks = (size + chunk_size - 1) / chunk_size;
+      bool check_valid = true;
```

Comment:
Please use atomic boolean here. Multi threads are trying to access to non atomic value